modules/tensor_mechanics/src/kernels/ALEKernel.C

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ALEKernel.h"
+
+template<>
+InputParameters validParams<ALEKernel>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Sets up derivetives with respect to initial configuration");
+  return params;
+}
+
+ALEKernel::ALEKernel(const InputParameters & parameters) :
+  Kernel(parameters),
+  _assembly_undisplaced(_fe_problem.assembly(_tid)),
+  _grad_phi_undisplaced(_assembly_undisplaced.gradPhi())
+{
+}
+
+void
+ALEKernel::computeJacobian()
+{
+  _fe_problem.prepareShapes(_var.number(), _tid);
```

Comment:
I have moved the prepareShapes from StressDivergence to here. Right now this class just sets up the shapes using undisplaced mesh. Should I remove this class altogether and make this call from StressDivergence?